modules/global/test/com/haulmont/bali/util/StringHelperTest.java

```diff
@@ -0,0 +1,37 @@
+package com.haulmont.bali.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author Dmitry Chigileychik
+ * @since 08.02.17.
+ */
+public class StringHelperTest {
+
+    @Test
+    public void testEmptyStringNormalize() {
+        String result = StringHelper.removeExtraSpaces("");
```

Comment:
It is failing.

> com.haulmont.bali.util.StringHelperTest > testEmptyStringNormalize FAILED
>     java.lang.StringIndexOutOfBoundsException at StringHelperTest.java:15

I wrote this test exactly to demonstrate this fail and show that in other cases your realisation and apache realisation works the same way